model/pod/pod.go

```diff
@@ -161,15 +187,30 @@ func (p *Pod) UpdateStatus(s Status) error {
 	byIDAndStatus := ByID(p.ID)
 	byIDAndStatus[StatusKey] = p.Status
 
+	updated := utility.BSONTime(time.Now())
+	setFields := bson.M{StatusKey: s}
+	switch s {
+	case StatusInitializing:
+		setFields[bsonutil.GetDottedKeyName(TimeInfoKey, TimeInfoInitializingKey)] = updated
+	case StatusStarting:
+		setFields[bsonutil.GetDottedKeyName(TimeInfoKey, TimeInfoStartingKey)] = updated
+	}
+
 	if err := UpdateOne(byIDAndStatus, bson.M{
-		"$set": bson.M{StatusKey: s},
+		"$set": setFields,
 	}); err != nil {
 		return err
 	}
 
 	event.LogPodStatusChanged(p.ID, string(p.Status), string(s))
```

Comment:
I feel like I asked this in a previous PR, but should `UpdateStatus()` verify the status has changed?

Reply:
I explained in the previous PR why I didn't do this, but since you seem to feel like it should be this way, I changed it to no-op if the status is identical.